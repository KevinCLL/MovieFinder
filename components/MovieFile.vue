<script>
export default {
  name: 'MovieFile',
  props: {
    movie: {
      type: Object,
      default: () => {},
    },
  },
  computed: {
    screenSize() {
      return this.$store.state.windowWidth
    },
    movieInfo() {
      const infoWantedAndSorted = [
        ['Title', this.movie.Title],
        ['Released', this.movie.Released],
        ['Director', this.movie.Director],
        ['Production', this.movie.Production],
        ['Writer', this.movie.Writer],
        ['Actors', this.movie.Actors],
        ['Country', this.movie.Country],
        ['Language', this.movie.Language],
        ['Genre', this.movie.Genre],
        ['Rated', this.movie.Rated],
        ['Awards', this.movie.Awards],
        ['BoxOffice', this.movie.BoxOffice],
        ['Plot', this.movie.Plot],
      ]
      return infoWantedAndSorted.filter((item) => this.itemExists(item[1]))
    },
  },
  methods: {
    itemExists(item) {
      return item && item !== 'N/A'
    },
  },
}
</script>

<template>
  <div
    class="flex w-full p-10 bg-gray-200 relative"
    :class="{ 'flex-col-reverse': screenSize < 763 }"
  >
    <div
      id="movieclose"
      class="cursor-pointer h-icon w-icon absolute right-0 top-0 mt-1 text-xl"
      @click="$emit('close')"
    >
      âœ–
    </div>
    <dl>
      <template v-for="item in movieInfo">
        <dt :key="`${item[0]}-t`">{{ item[0] }}:</dt>
        <dd :key="`${item[0]}-d`">{{ item[1] }}</dd>
      </template>
    </dl>
    <div class="flex flex-col min-w-1/4">
      <div v-if="itemExists(movie.Poster)" class="my-0 mx-auto">
        <img :src="movie.Poster" :alt="movie.Title" />
      </div>
      <div v-if="movie.Ratings && movie.Ratings.length > 0">
        <div v-for="rating in movie.Ratings" :key="rating.Source">
          {{ rating.Source }} : {{ rating.Value }}
        </div>
      </div>
      <div
        v-if="itemExists(movie.imdbID)"
        class="cursor-pointer bg-yellow-400 mb-4 text-center"
      >
        <a :href="`https://www.imdb.com/title/${movie.imdbID}`" target="_blank">
          View IMDB
        </a>
      </div>
    </div>
  </div>
</template>

<style scoped>
dl {
  width: 100%;
  overflow: hidden;
  padding: 0;
  padding-right: 10px;
  margin: 0;
}
dt {
  font-weight: bold;
  float: left;
  text-align: right;
  width: 90px;
  padding-right: 10px;
  margin: 0;
}
dd {
  float: left;
  text-align: left;
  width: 80%;
  padding: 0;
  margin: 0;
}
@media (max-width: 540px) {
  dt {
    text-align: left;
    width: 100%;
  }
  dd {
    width: 100%;
  }
}
</style>
