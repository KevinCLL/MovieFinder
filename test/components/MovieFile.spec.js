import { shallowMount, createLocalVue } from '@vue/test-utils'
import MovieFile from '@/components/MovieFile.vue'
import Vuex from 'vuex'
import storeConfig from '../store-config'
import movie from '../data/detailedMovie.js'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('MovieFile', () => {
  let wrapper
  let store

  beforeEach(() => {
    store = new Vuex.Store(storeConfig)
    wrapper = shallowMount(MovieFile, {
      localVue,
      propsData: {
        movie,
      },
      store,
    })
  })
  afterEach(() => {
    if (wrapper) {
      wrapper.destroy()
    }
  })

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
