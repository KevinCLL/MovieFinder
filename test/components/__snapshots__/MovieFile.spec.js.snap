// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieFile renders correctly 1`] = `
<div
  class="flex w-full p-10 bg-gray-200 relative"
>
  <div
    class="cursor-pointer h-icon w-icon absolute right-0 top-0 mt-1 text-xl"
    id="movieclose"
  >
    
    âœ–
  
  </div>
   
  <dl>
    <dt>
      Title:
    </dt>
     
    <dd>
      Primer
    </dd>
    <dt>
      Released:
    </dt>
     
    <dd>
      27 May 2005
    </dd>
    <dt>
      Director:
    </dt>
     
    <dd>
      Shane Carruth
    </dd>
    <dt>
      Production:
    </dt>
     
    <dd>
      ThinkFilm
    </dd>
    <dt>
      Writer:
    </dt>
     
    <dd>
      Shane Carruth
    </dd>
    <dt>
      Actors:
    </dt>
     
    <dd>
      Shane Carruth, David Sullivan, Casey Gooden, Anand Upadhyaya
    </dd>
    <dt>
      Country:
    </dt>
     
    <dd>
      USA
    </dd>
    <dt>
      Language:
    </dt>
     
    <dd>
      English, French
    </dd>
    <dt>
      Genre:
    </dt>
     
    <dd>
      Drama, Sci-Fi, Thriller
    </dd>
    <dt>
      Rated:
    </dt>
     
    <dd>
      PG-13
    </dd>
    <dt>
      Awards:
    </dt>
     
    <dd>
      3 wins & 7 nominations.
    </dd>
    <dt>
      BoxOffice:
    </dt>
     
    <dd>
      $392,420
    </dd>
    <dt>
      Plot:
    </dt>
     
    <dd>
      Engineers Aaron, Abe, Robert and Phillip are working on an invention, the prototype being built in Aaron's garage. This project is beyond their day jobs. The project truly does belong to Aaron and Abe, as they use all their free time working on it, primarily trying to overcome the many engineering related problems they've encountered. It is during one of his tests with the invention running that Abe discovers that a protein inside the main unit has multiplied much more rapidly than it could in nature. Rather than the invention being a protein super incubator, Abe, using himself as a guinea pig, and a very meticulous one at that, discovers that the invention can be used as a time machine. In his self experiment, Abe was especially careful not to interfere with his own self in that time warp. Abe passes along this discovery to Aaron, who he expects will tell his wife Kara in what is the sanctity of their marriage, but he doesn't want to tell either Robert or Phillip. Much to Abe's surprise, Aaron does not want to tell Kara, it being a sole intellectual property of just the two of them, in the process moving the base of operation to a locked storage unit. Aaron's plan for the two of them is to use the invention to win big in the stock market by knowing through the time travel what has happened in the market. With two thoughts on the matter now instead of just one, Aaron and Abe may hit some logistical and philosophical roadblocks in how to move forward.
    </dd>
  </dl>
   
  <div
    class="flex flex-col min-w-1/4"
  >
    <div
      class="my-0 mx-auto"
    >
      <img
        alt="Primer"
        src="https://m.media-amazon.com/images/M/MV5BNjc3OWVjMTItYjc0Yi00NmFlLTk2YTgtYmU0MzcxMjBkNTYxXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg"
      />
    </div>
     
    <div>
      <div>
        
        Internet Movie Database : 6.9/10
      
      </div>
      <div>
        
        Rotten Tomatoes : 73%
      
      </div>
      <div>
        
        Metacritic : 68/100
      
      </div>
    </div>
     
    <div
      class="cursor-pointer bg-yellow-400 mb-4 text-center"
    >
      <a
        href="https://www.imdb.com/title/tt0390384"
        target="_blank"
      >
        
        View IMDB
      
      </a>
    </div>
  </div>
</div>
`;
